### PR TITLE
[XM] Add release value option to msbuild/mmp to resolve XM 4.5 assemb…

### PR DESCRIPTION
**Pull request by chamons**
…lies from system GAC

- This option "reverts" a C7 fix that prevented resovling assemblies from the GAC, which is unsafe
- If you use this option, you need to know what you are doing. The mono BCL and the XM BCL need to be compatible
- Use strictly puts you in the no support "you get to keep the pieces if it breaks" category

### CODE REVIEW

**Comment by chamons** [comment]
catagory -> category in the commit message typo

**Comment by xamarin-release-manager** [comment]
Build failure

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -284,6 +284,7 @@ static void Main2 (string [] args)
 				{ "tls-provider=", "Specify the default TLS provider", v => { tls_provider = v; }},
 				{ "http-message-handler=", "Specify the default HTTP Message Handler", v => { http_message_provider = v; }},
 				{ "extension", "Specifies an app extension", v => is_extension = true },
+				{ "allow_unsafe_gac_resolution", "Allow msbuild to resolve from the System GAC", v => {} , true }, // Used in Xamarin.Mac.XM45.targets and must be ignored here. Hidden since it is a total hack. If you can use it, you don't need support
```
Comment (abock) [inline tools/mmp/driver.cs]: `allow-unsafe-gac-resolution` would be more inline with other options

`msbuild` → `MSBuild`

Comment (abock) [inline tools/mmp/driver.cs]: Maybe `unsafe` → `unsupported` ?

**Comment by abock** [comment]
👍

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**